app/src/main/java/com/ahgitdevelopment/course/customexamples/features/screens/datastore/timer/TimerViewModel.kt

```diff
@@ -0,0 +1,87 @@
+package com.ahgitdevelopment.course.customexamples.features.screens.datastore.timer
+
+import android.os.CountDownTimer
+import android.os.SystemClock
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.ahgitdevelopment.course.customexamples.repository.local.timer.TimerRepositoryImpl
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+private const val SECOND: Long = 1_000L
```

Comment:
I think this is correct but I rather have this line inside a companion object instead. 
It's up to you

Reply:
a complement object is called if i have call it in other classes, sure?

Reply:
What do you mean?